### PR TITLE
Migrate existing flow typed files to typescript

### LINKED ISSUE #84: [TypeScript] Convert existing types annotated w/ Flow to TypeScript
Child of #43

- [x] Update babel configuration - strip TypeScript assertions instead of Flow.
- [x] Remove Flow stubs.

Existing files

- [x] src/lib/apollo/resolvers/localNetwork.js
- [x] src/lib/buildInfo.js
- [x] src/lib/component/util/NavigationProvider/NavigationProvider.js
- [x] src/lib/component/util/NavigationProvider/WithNavigation.js
- [x] src/lib/component/util/NavigationProvider/context.js
- [x] src/lib/component/util/NavigationProvider/index.js
- [x] src/lib/component/util/Query.js
- [x] src/lib/error/QueryAbortedError.js
- [x] src/lib/error/FetchError.js
- [x] src/lib/util/compose.js
- [x] src/lib/util/fetch.js

### PR DESCRIPTION
**Pull request by 10xjs**
Signed-off-by: Neal Granger <neal@nealg.com>

## What is this change?

Replace Flow and related tooling with typescript. Convert any existing flow declarations to typescript.

## Why is this change necessary?

Closes #84

This unblocks the ability to begin writing new code in typescript and migrating existing unttyped code.

## Does your change need a Changelog entry?

No.

## Do you need clarification on anything?

No.

## Were there any complications while making this change?

Typescript has a unique "import assignment" syntax for importing commonjs modules which differs from babel's "interop require" behavior.

Canonical Typescript:
```js
// importing
import React = require("react");

// re-exporting
export =  React;
```

Es-module:
```ts
// importing
import * as React from "react";

// re-exporting
export * from "react";
```

`react` does not truly have named or default exports however babel allows referencing properties of the `module.exports` object as if they were es-module named exports.

Typescript in `esModuleInterop` mode does the same. However typescript does not allow re-exporting the declared common-js module types without using the export assignment syntax which babel cannot parse.

The workaround that I have come up with is to use an untyped js file to define the babel-parseable exports and a matching `.d.ts` declaration file which describes the exports of the untyped file.

Unfortunately the obvious solution doesn't work:
```ts
import React = require("react");
export = React;
```
Typescript still complains with "This module can only be referenced with ECMAScript imports/exports" when loading the `d.ts` file.

This appears to work instead:
```ts
/// <reference path="node_modules/@types/react/index.d.ts" />
export = React;
```
## Have you reviewed and updated the documentation for this change? Is new documentation required?

No.

## How did you verify this change?

The app builds and runs as it should.

### CODE REVIEW

File: type/deepmerge.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+declare module 'deepmerge' {
```
Comment (jamesdphillips) [inline type/deepmerge.d.ts]: Unless I misunderstand, does `@types/deepmerge` not already provide this?

Comment (10xjs) [reply type/deepmerge.d.ts]: I was having trouble with `@types/deepmerge` 
https://www.npmjs.com/package/@types/deepmerge

I will experiment a bit more to see if the statement: "deepmerge provides its own type definitions" is true.

File: src/lib/component/base/RelativeDate/RelativeDate.test.js [hunk 2]
```diff
@@ -1,40 +1,47 @@
...
 test("RelativeDate given date seconds ago", () => {
   const now = new Date();
   const date = new Date(now.getTime() - 12000);
-  const elem = mount(<RelativeDate to={now} dateTime={date.toUTCString()} />);
+  const { getByText } = render(
+    <RelativeDate to={now} dateTime={date.toUTCString()} />,
+  );
 
-  expect(elem.text()).toEqual("seconds ago");
+  expect(getByText("seconds ago"));
```
Comment (jamesdphillips) [inline src/lib/component/base/RelativeDate/RelativeDate.test.js]: This seems a bit strange, do we not want to have it match a value?

File: src/lib/component/partial/EventsList/StatusMenu.js [hunk 3]
```diff
@@ -15,23 +15,33 @@ import { ErrorHollowIcon } from "/lib/component/icon";
 const options = [
   {
     value: "incident",
-    Icon: () => <ErrorHollowIcon />,
+    Icon: function IncidentIcon() {
```
Comment (jamesdphillips) [inline src/lib/component/partial/EventsList/StatusMenu.js]: Is this a TypeScript convention?

Comment (10xjs) [reply src/lib/component/partial/EventsList/StatusMenu.js]: This is to satisfy an eslint rule that asserts components have display names.

File: src/lib/component/base/RelativeDate/RelativeDate.test.js (hunk 2 above)
Comment (10xjs) [reply src/lib/component/base/RelativeDate/RelativeDate.test.js]: I think I can improve the matcher I'm using here.

**Pull request merged**